src/zenohUTransport.cpp

```diff
@@ -289,28 +291,37 @@ UCode ZenohUTransport::sendQueryable(const UUri &uri,
         options.encoding = z_encoding(Z_ENCODING_PREFIX_TEXT_PLAIN, nullptr);
     }
 
-    auto message = MessageBuilder::build(attributes, payload);
-    if (0 == message.size()) {
-        spdlog::error("MessageBuilder::instance().build failed");
-        return UCode::INTERNAL;
-    }
+    auto header = MessageBuilder::buildHeader(attributes);
+
+    size_t headerLength = header.size();
+    const uint8_t* headerPointer = header.data();
+
+    z_owned_bytes_map_t map = z_bytes_map_new();
+    options.attachment = z_bytes_map_as_attachment(&map);
+
+
+    z_bytes_t headerBytes = {.len = headerLength, .start = headerPointer};
+    z_bytes_map_insert_by_alias(&map, z_bytes_new("header"), headerBytes);
 
     z_query_t lquery = z_loan(query);
 
-    if (0 != z_query_reply(&lquery, z_query_keyexpr(&lquery), message.data(), message.size(), &options)) {
+    if (0 != z_query_reply(&lquery, z_query_keyexpr(&lquery), nullptr, 0, &options)) {
```

Comment:
why did you put it to nullptr? it would crash - did you test the code?
you should send the payload (the reply)